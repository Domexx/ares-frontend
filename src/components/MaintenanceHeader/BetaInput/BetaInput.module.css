/* === === === === BetaInput === === === === */

.BetaInput {
  display: flex;
  flex-direction: column;
  grid-column: 5 / span 6;
  gap: 30px;
  text-align: center;
}

.BetaInput-description {
  padding: 0 50px 0 50px;
  font-size: 12px;
}

.BetaInput-form {
  display: flex;
  gap: 15px;
  justify-content: space-between;
}

.BetaInput-button {
  width: 200px;
}

.BetaInput-element {
  width: 100%;
  padding: 25px;
  border: none;
  border-radius: 4px;
  background: var(--clr-dark-400);
  color: var(--clr-text-primary);
  outline: none;
}

.BetaInput-info {
  font-size: 12px;
}

@media (--tablet-landscape) {
  .BetaInput {
    grid-column: 3 / span 12;
  }
}

@media (--tablet) {
  .BetaInput {
    grid-column: 2 / span 14;
  }
}

@media (--phone) {
  .BetaInput-headline {
    margin-top: 30px;
  }

  .BetaInput-description {
    padding: 0;
  }

  .BetaInput-form {
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .BetaInput-button {
    width: 100%;
  }
}
