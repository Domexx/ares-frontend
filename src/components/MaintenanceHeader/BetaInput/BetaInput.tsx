import { useFormik } from 'formik';
import useTranslation from 'next-translate/useTranslation';
import React from 'react';
import toast from 'react-hot-toast';
import Button from '../../../elements/Button';
import s from './BetaInput.module.css';

export type Props = {
  headline: string;
  description: string;
  ctaText: string;
};

const BetaInput: React.FC<Props> = ({ headline, description, ctaText }) => {
  const { t } = useTranslation('maintenance');

  const form = useFormik({
    initialValues: {
      code: '',
    },
    onSubmit: () => {
      toast.error(t('not_available'));
    },
  });

  return (
    <div className={s['BetaInput']}>
      <h1 className={`${s['BetaInput-headline']} title`}>{headline}</h1>
      <p className={s['BetaInput-description']}>{description}</p>
      <form className={s['BetaInput-form']} onSubmit={form.handleSubmit}>
        <input
          disabled
          className={s['BetaInput-element']}
          id="code"
          name="code"
          placeholder={t('insert_code')}
          type="text"
          onChange={form.handleChange}
          value={form.values.code}
        />
        <Button className={s['BetaInput-button']}>{ctaText}</Button>
      </form>
      <p className={s['BetaInput-info']}>{t('not_enabled')}</p>
    </div>
  );
};

export default BetaInput;
