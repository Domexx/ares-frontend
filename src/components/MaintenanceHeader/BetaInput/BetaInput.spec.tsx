import React from 'react';

import { render } from '@testing-library/react';

import BetaInput from './BetaInput';

describe('Beta Input Component', () => {
  it('should render properly', () => {
    const { container } = render(
      <BetaInput headline="TestHeadline" description="Test description" ctaText="click here" />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
