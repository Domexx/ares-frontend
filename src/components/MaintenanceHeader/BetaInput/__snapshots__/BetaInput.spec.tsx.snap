// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beta Input Component should render properly 1`] = `
<div
  class="BetaInput"
>
  <h1
    class="BetaInput-headline title"
  >
    TestHeadline
  </h1>
  <p
    class="BetaInput-description"
  >
    Test description
  </p>
  <form
    class="BetaInput-form"
  >
    <input
      class="BetaInput-element"
      disabled=""
      id="code"
      name="code"
      placeholder="insert_code"
      type="text"
      value=""
    />
  </form>
  <p
    class="BetaInput-info"
  >
    not_enabled
  </p>
</div>
`;
